### DISCUSSION TITLE
How to differentiate between style and syntax errors?

### DISCUSSION THREAD

**Discussion by pstahlhofen** [w0rp/ale#4596] [Q&A - Ask for help with problems] [answered]
I am new to ALE and my main goal for using this plugin is that I want to see syntax and name errors (such as undefined variables) in my Python code before actually running it. Adjusting my code according to style guides is not my main priority. In fact, some style recommendations (like the insertion of blank lines at many locations) might even hinder me because I have a visual impairment, hence I have to use a magnifier and only small portions of code fit on my screen at once.
I would like to find a way to fix ONLY the real errors and to make ALE showing me only those. This leads me to the following questions:
1. Is there a linter for Python that shows only syntax errors? If yes, how would I set up ALE to use this linter only.
2. If no such linter exists, could I navigate the error list of ALE and navigate automatically only to the places in my code where real errors occurred?
Any answers would be very appreciated

**Comment by w0rp** [comment, marked as answer]
ALE linters can add a `subtype` key value for `style` to indicate if errors are warnings are for style or not. Wherever you see they don't, open issues for them or create pull requests. Check out `:help ALEStyleError`.

For disabling particular errors or warnings, configure the tool itself to do that with its own project files.

**Reply by pstahlhofen** [reply]
Alright, tank you!

**Reply by pstahlhofen** [reply]
@w0rp Could you tell me how to make ALE only highlight the errors that are not ALEStyleErrors? Is it possible to add only those errors to the loc-list?

**Reply by pstahlhofen** [reply]
@w0rp Okay, it seems like I found a solution now, so I'll write it down here for anyone else who is also looking for a linter only showing "real errors" for Python. Here's a step by step guide:
1. Install `pylint` (e.g. via `conda install pylint`)
2. Go to your `~/.vimrc` and add the following lines:
```
let g:ale_linters={'python': ['pylint']}
let g:ale_linters_explicit=1
```
Those will make sure that ALE uses `pylint` and nothing else
3. Go to the place where you installed ale (in my case `~/.vim/pack/my_plugins/start`). Navigate to `ale/ale_linters/python` and edit the file `pylint.vim`. Change the line
```
call ale#Set('python_pylint_options', '')
```
to
```
call ale#Set('python_pylint_options', '-E')
```
From now on, `pylint` will always be called with the `-E` option, making it only show real errors.
